08_education/preschool_county.qmd

```diff
@@ -0,0 +1,766 @@
+
+```
+
+```{r}
+rm(acs_race_ethnicity)
+```
+
+## Add Suppressed Rows to Data
+
+To create the metric we suppressed counties that did not meet the threshold for sample size but we need to include these in the final data. Create an all version of the data that includes counties that were suppressed prior to the survey_mean calculation. 
+
+Create expanded version for race-ethnicity.
+```{r}
+results_race_ethnicity_expanded <- acs_crosswalked %>% 
+  filter(sample_structure == "5year") %>% 
+  mutate(geoid = paste0(statefip, county)) %>% 
+  group_by(year, geoid, subgroup_race) %>% 
+  count() %>% 
+  expand_grid() %>%  
+   mutate(statefip = str_sub(geoid, 1, 2),
+         county = str_sub(geoid, 3, 5)) %>% 
+  left_join(results_race_ethnicity, by = c("year", "statefip", "county", "subgroup_race")) %>% 
+  mutate(subgroup_type = "race-ethnicity") %>% 
+  rename(subgroup = subgroup_race)
+```
+
+## Data Quality Flags
+
+Add a flag for data quality, this is a numeric variable between 1 and 3 with 1 representing the best quality and 3 representing the worst. 
+
+First combine the subgroup data sets.
+```{r}
+metrics_income_sub <-results_all_expand %>% 
+  mutate(subgroup = "All",
+         subgroup_type = "all") %>% 
+  bind_rows(results_race_ethnicity_expanded)
+```
+
+First, join on to county_puma which provides the quality of the crosswalk (`afact * afact2`).
+```{r}
+
+metrics_education_sub <- left_join(metrics_income_sub, county_puma, by=c("crosswalk_period", "statefip","county"))
+```
+
+Now create the metric quality variable. The `share_in_preschool_quality` variable combines information on the quality of the crosswalk and the sample size (effective sample) to create a final quality flag. Note that any county with NA for the `share_in_preschool` metric is being suppressed due to sample size and is given a quality flag of 3.  
+
+```{r}
+metrics_education_sub <- metrics_education_sub %>% 
+         mutate(share_in_preschool_quality = case_when(is.na(share_in_preschool) ~ 3,
+                                                   geographic_allocation_quality == 1 ~ 1,
+                                                   geographic_allocation_quality == 2 ~ 2,
+                                                   geographic_allocation_quality == 3 ~ 3)
+  )
+```
+
+### Validation
+
+Summarize the predictors for the 1-year and 5-year data.
+```{r}
+metrics_education_all %>% 
+  select(share_in_preschool:share_in_preschool_quality) %>% 
+  summary()
+
+metrics_education_sub %>% 
+  select(share_in_preschool:share_in_preschool_quality) %>% 
+  summary()
+```
+
+Look at distribution of prek_attendance share by county.
+```{r}
+
+metrics_education_all %>% 
+  filter(year == 2021) %>% 
+  select(share_in_preschool) %>% 
+  ggplot(aes(x = share_in_preschool)) +
+  geom_density(alpha = 0.15) +
+  theme_minimal() +
+  ggtitle("Distribution of share prek_attendance, by county 2021 (5-year data)") +
+  ylab("Density")
+
```

Comment:
Instead of creating five different visualizations. Create one and facet. 

```r
metrics_education_sub %>% 
  filter(year == 2021) %>% 
  ggplot(aes(x = share_in_preschool)) +
  geom_density(alpha = 0.15) +
  facet_wrap(~subgroup) +
  labs(
    title = "Distribution of share prek_attendance, by county 2021 (5-year data)",
    y = "Density"
  ) +
  theme_minimal()

```